### PR TITLE
Update go apis and go version

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Comment by iamniting** [comment]
@SanjalKatiyar Can you look at the second commit? The v1alpha package is removed from the API, so I started using v1.

```
$ go mod tidy
go: finding module for package github.com/openshift/api/console/v1alpha1
go: downloading github.com/openshift/api v0.0.0-20240702164632-a89a7c1e20ad
go: github.com/red-hat-storage/odf-operator imports
	github.com/openshift/api/console/v1alpha1: module github.com/openshift/api@latest found (v0.0.0-20240702164632-a89a7c1e20ad), but does not contain package github.com/openshift/api/console/v1alpha1
```

**Comment by SanjalKatiyar** [comment]
> @SanjalKatiyar Can you look at the second commit? The v1alpha package is removed from the API, so I started using v1.
> 
> ```
> $ go mod tidy
> go: finding module for package github.com/openshift/api/console/v1alpha1
> go: downloading github.com/openshift/api v0.0.0-20240702164632-a89a7c1e20ad
> go: github.com/red-hat-storage/odf-operator imports
> 	github.com/openshift/api/console/v1alpha1: module github.com/openshift/api@latest found (v0.0.0-20240702164632-a89a7c1e20ad), but does not contain package github.com/openshift/api/console/v1alpha1
> ```

AFAIK, this is the correct syntax of v1 API (`spec.backend.service` & `proxy[<INDEX>].authorization` etc):
```
apiVersion: console.openshift.io/v1
kind: ConsolePlugin
metadata:
  name: odf-console
spec:
  backend:
    service:
      basePath: /
      name: odf-console-service
      namespace: openshift-storage
      port: 9001
    type: Service
  displayName: ODF Plugin
  i18n:
    loadType: ''
  proxy:
    - alias: provider-proxy
      authorization: UserToken
      endpoint:
        service:
          name: ux-backend-proxy
          namespace: openshift-storage
          port: 8888
        type: Service
    - alias: rosa-prometheus
      authorization: None
      endpoint:
        service:
          name: prometheus
          namespace: openshift-storage
          port: 9339
        type: Service
```
But not sure why you did not get any error !!

**Comment by iamniting** [comment]
> > @SanjalKatiyar Can you look at the second commit? The v1alpha package is removed from the API, so I started using v1.
> > ```
> > $ go mod tidy
> > go: finding module for package github.com/openshift/api/console/v1alpha1
> > go: downloading github.com/openshift/api v0.0.0-20240702164632-a89a7c1e20ad
> > go: github.com/red-hat-storage/odf-operator imports
> > 	github.com/openshift/api/console/v1alpha1: module github.com/openshift/api@latest found (v0.0.0-20240702164632-a89a7c1e20ad), but does not contain package github.com/openshift/api/console/v1alpha1
> > ```
> 
> AFAIK, this is the correct syntax of v1 API (`spec.backend.service` & `proxy[<INDEX>].authorization` etc):
> 
> ```
> apiVersion: console.openshift.io/v1
> kind: ConsolePlugin
> metadata:
>   name: odf-console
> spec:
>   backend:
>     service:
>       basePath: /
>       name: odf-console-service
>       namespace: openshift-storage
>       port: 9001
>     type: Service
>   displayName: ODF Plugin
>   i18n:
>     loadType: ''
>   proxy:
>     - alias: provider-proxy
>       authorization: UserToken
>       endpoint:
>         service:
>           name: ux-backend-proxy
>           namespace: openshift-storage
>           port: 8888
>         type: Service
>     - alias: rosa-prometheus
>       authorization: None
>       endpoint:
>         service:
>           name: prometheus
>           namespace: openshift-storage
>           port: 9339
>         type: Service
> ```
> 
> But not sure why you did not get any error !!

Actually I got the errors, updated the code, Can you pls check it again.

File: console/console.go [hunk 1]
```diff
@@ -81,42 +80,51 @@ func GetService(port int, namespace string) *apiv1.Service {
...
+			Endpoint: consolev1.ConsolePluginProxyEndpoint{
+				Type: consolev1.ProxyTypeService,
+				Service: &consolev1.ConsolePluginProxyServiceConfig{
+					Name:      "ux-backend-proxy",
+					Namespace: serviceNamespace,
+					Port:      8888,
+				},
 			},
-			Authorize: true,
+			Authorization: consolev1.None,
```
Comment (SanjalKatiyar) [inline console/console.go:95]: `Authorization: consolev1.UserToken`

**Review by SanjalKatiyar — COMMENTED** [review]
LGTM

File: console/console.go (hunk 1 above)
Comment (iamniting) [reply console/console.go]: Done

**Review by SanjalKatiyar — APPROVED** [review]

**Pull request merged**